### PR TITLE
Adds construct / deconstruct crates

### LINKED ISSUE #4993: Make crates and lockers craftable
What title says

### PR DESCRIPTION
**Pull request by Peptide90**
Added construction and deconstruction for non secure crates. Also fixed construction graph for livestock crate. 

Also added a construction condition for checking if a crate is welded - more work is needed on this to make it a true false condition so it can actually be used to make sure a crate is NOT welded.

Also we need a condition for checking if a storage is locked so we can make graphs for secure storage. Will open issue request.

Closes #4993

Will do lockers next but split the PR into two parts because it took longer than expected to make all the files.

:cl:
- add: Added construction for main crates and deconstruction for all crates. Use a screwdriver to deconstruct crates.

### CODE REVIEW

**Comment by Peptide90** [comment]
Welp, tests are a lie. Works fine locally no errors. Only error the test is throwing is a missing key for something that doesn't exist and there is no reference to in the code anymore...

**Comment by mirrorcult** [comment]
ok this is way too many crates. ss13 only lets you build one kind

**Comment by Peptide90** [comment]
> ok this is way too many crates. ss13 only lets you build one kind

Removed all other crate entries except plastic, metal and livestock so we have the three material options for crates at least. Left the graphs as is so the deconstruction still works. (Didn't want to just comment them out as the file size was 3kb bigger with them all in but up to you guys if you want them left commented out instead).

When I get more time monday I'll port a regular wooden crate sprite over and replace the construction entry for the livestock crate with it so we have a generic wooden crate - will leave livestock crate in of course for cargo orders.

**Comment by Peptide90** [comment]
Hubismal has done a condition for checking if a container is locked and I have tested it so will now expand this PR to include deconstruction for all crates.

Will still do lockers in a separate PR since I've not started them yet.

**Comment by Peptide90** [comment]
Tests will fail until prerequesit PR merged.

File: Resources/Prototypes/Recipes/Crafting/Graphs/cratesecgear.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          conditions:
+            - !type:StorageWelded
+              welded: false
+            - !type:Locked
+              locked: false              
+          completed:
+            - !type:SpawnPrototype
+              prototype: SheetSteel1
+              amount: 5
+            - !type:DeleteEntity {}
```
Comment (mirrorcult) [inline Resources/Prototypes/Recipes/Crafting/Graphs/cratesecgear.yml:30]: theres still a fuck ton of duplication across all these, i get why its done but at the very least put all of these in their own folder

File: Resources/Prototypes/Recipes/Crafting/crates.yml [hunk 2]
```diff
@@ -7,5 +7,26 @@
...
   icon: Structures/Storage/Crates/livestock.rsi/livestockcrate.png
-  objectType: Item
-  
\ No newline at end of file
+  objectType: Structure
+  
+- type: construction
+  name: metal crate
+  id: CrateGenericonimo
+  graph: CrateGenericonimo
```
Comment (mirrorcult) [inline Resources/Prototypes/Recipes/Crafting/crates.yml:15]: genericonimo?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Crafting/crates.yml (hunk 2 above)
Comment (Peptide90) [reply Resources/Prototypes/Recipes/Crafting/crates.yml:15]: Apparently thats the ID for the generic crate, beyond me but I can change it throughout if that's better.

File: Resources/Prototypes/Recipes/Crafting/Graphs/cratesecgear.yml (hunk 1 above)
Comment (Peptide90) [reply Resources/Prototypes/Recipes/Crafting/Graphs/cratesecgear.yml:30]: Wasn't sure if sticking all the crates in their own construction folder would still function. Will do that if that's alright.

Again wasn't sure if having one graph that applied to the crate base entity would work across all crates but when I tested with lockers it didn't work.

**Comment by Peptide90** [comment]
Closes #5667, code duplication fixed by reducing the amount of graphs to 4 (one for steel, plastic, wood and secure crates) and reducing the amount of actually craftable crates to steel, wood and plastic so we cover all bases of available resources.

5667 is resolved thanks to a condition by @hubismal which he has provided for this PR to be complete.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**